File: C/Lab4.cpp
#include <iostream>
using namespace std;
int main()
{
	string option;
	option = "Si"; //Opción para saber si se repite o no
	int entrada;

	bool a = false;
	bool b = false;
	bool c =false;
	bool d = false;
	cout<<"Ingresa 1 si la maquina envio alerta o 0 si no\n";
	while(option == "Si")
	{
		for(int i = 0; i<4; i++) //Ingreso de datos
		{
			cout<<"\nIngrese datos de la maquina ["<<i+1<<"]: ";
			cin>>entrada;
			switch (i)
			{
				case 0:
					if(entrada == 1)
					{
						a = true;
					}
					else
					{
						a = false;
					}
				case 1:
					if(entrada == 1)
					{
						b = true;
					}
					else
					{
						b = false;
					}
				case 2:
					if(entrada == 1)
					{
						c = true;
					}
					else
					{
						c = false;
					}
				case 3:
					if(entrada == 1)
					{
						d = true;
					}
					else
					{
						d = false;
					}
			}
		}
		if((a==true)||(c==true)) //maquiana S1 alerta
		{
			cout<<"\n S1 esta en peligro";
		}
		else
		{
			cout<<"\n S1 no esta en peligro";
		}
		if((c && (a || b || d))||(a && b &&  d)) //maquiana S2 alerta
		{
			cout<<"\n S2 esta en peligro";
		}
		else
		{
			cout<<"\n S2 no esta en peligro";
		}
		if((a==true)||(d && (c || b))) //maquiana S3 alerta
		{
			cout<<"\n S3 esta en peligro";
		}
		else
		{
			cout<<"\n S3 no esta en peligro";
		}
	}
	
	
	return 0;
}

File: Python/lab4.py
import datetime
hora_actual = datetime.datetime.now()
#Atributos a usar
a = False
b = False
c = False
d = False
cont = 1
#Funciones
def booleano(a):
    if(a == 1):
        return True
    return False
def activar_alarma_s1():
    if a or c:
        print("\nAlarma S1 activada","\nLa hora emitida de la alarma:", hora_actual)
    else:
        print("\nS1 esta a salvo")
def activar_alarma_s2():
    if (c and(a or b or d)) or (a and b and d):
        print("\nAlarma S2 activada","\nLa hora emitida de la alarma:", hora_actual)
    else:
        print("\nS2 esta a salvo")
def activar_alarma_s3():
    if ((d and (c or b)) or (a)):
        print("\nAlarma S3 activada","\nLa hora emitida de la alarma:", hora_actual)
    else:
        print("\nS3 esta a salvo")
#Ingreso de datos
print("\nSistema de alarmas\n")
for i in range(4):
    try:
        while True:
            x = int(input("Ingrese el dato de la maquina: "))   
            if(x==0) or (x==1):
                break
            else:
                print("\nIngrese un valor entre 1 y 0\n")
    except ValueError:
        print("opción invalida")
    if(i == 0):
        a = booleano(x)
    if(i == 1):
        b = booleano(x)
    if(i == 2):
        c = booleano(x)
    if(i == 3):
        d = booleano(x)
activar_alarma_s1()
activar_alarma_s2()
activar_alarma_s3()